src/dynamo0p3_builtins.py

```diff
@@ -327,14 +360,35 @@ def __str__(self):
     def gen_code(self, parent):
         from f2pygen import AssignGen
         # We multiply each element of f1 by the scalar argument and
-        # store the result in f2
+        # store the result in f2.
         scalar_name = self._arguments.args[0].name
         invar_name = self.array_ref(self._arguments.args[1].proxy_name)
         outvar_name = self.array_ref(self._arguments.args[2].proxy_name)
         parent.add(AssignGen(parent, lhs=outvar_name,
                              rhs=scalar_name + " * " + invar_name))
 
 
+class DynAXMYKern(DynBuiltIn):
+    ''' f = a.x - y where 'a' is a scalar and 'f', 'x' and
+    'y' are fields '''
+
+    def __str__(self):
+        return "Built-in: AXMY"
+
+    def gen_code(self, parent):
+        from f2pygen import AssignGen
+        # We multiply one element of field f1 (2nd arg) by a scalar
+        # (1st arg), subtract it from the corresponding
+        # element of a second field (3rd arg)  and write the value to the
+        # corresponding element of field f3 (4th arg).
+        scalar_name = self._arguments.args[0].name
+        invar_name1 = self.array_ref(self._arguments.args[1].proxy_name)
+        invar_name2 = self.array_ref(self._arguments.args[2].proxy_name)
+        outvar_name = self.array_ref(self._arguments.args[3].proxy_name)
+        rhs_expr = scalar_name + "*" + invar_name1 + " - " + invar_name2
+        parent.add(AssignGen(parent, lhs=outvar_name, rhs=rhs_expr))
+
+
```

Comment:
Quite right, inc_axpy, axpy, and axmy do as one would expect. I must have read it incorrectly before, apologies.